api/kyverno/v1/image_verification_types.go

```diff
@@ -236,9 +236,14 @@ type CTLog struct {
 // OCI registry and decodes them into a list of Statements.
 type Attestation struct {
 	// PredicateType defines the type of Predicate contained within the Statement.
-	// +kubebuilder:validation:Required
+	// Deprecated in favour of 'Type', to be removed soon
+	// +kubebuilder:validation:Optional
 	PredicateType string `json:"predicateType" yaml:"predicateType"`
 
+	// Type defines the type of attestation contained within the Statement.
+	// +kubebuilder:validation:Optional
+	Type string `json:"type" yaml:"type"`
```

Comment:
Can we add a validation check that one of these two fields needs to be specified with the imageVerify type of rule? And print a warning that the `predicateType` has been deprecated?

Reply:
I am doing it like this

deprecated message: https://github.com/Vishal-Chdhry/kyverno/blob/1e0d453aeaaee8bcfcbbda5bba1c6e1f4ae2769d/pkg/engine/internal/imageverifier.go#L481-L486
Is there a way to log a message in the terminal saying that predicate type is deprecate, when the policy is applied?

To check that one of the two fields are set: https://github.com/Vishal-Chdhry/kyverno/blob/1e0d453aeaaee8bcfcbbda5bba1c6e1f4ae2769d/pkg/engine/internal/imageverifier.go#L334-L336

Reply:
>Is there a way to log a message in the terminal saying that predicate type is deprecate, when the policy is applied?

You can return the message as a warning along with the admission response.